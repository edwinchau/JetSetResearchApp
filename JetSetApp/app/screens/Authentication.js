import React, { Component } from 'react';
import {
    ActivityIndicator,
    StatusBar,
    StyleSheet,
    View,
} from 'react-native';

import * as FileSystem from "expo-file-system";

styles = StyleSheet.create({
    background: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'white',
    },
    container: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'white',
        elevation: 20,
        borderRadius: 10,
    },
    questionText: {
        marginBottom: 20,
        fontSize: 20
    },
    headerStyle: {
        backgroundColor: 'rgba(106,177,135,1)',
        height: 40,
        elevation: 5,
    },
});

export default class Authentication extends Component {
    static navigationOptions = () => {
        return {
            headerStyle: styles.headerStyle,
            headerTintColor: '#fff',
            headerTitle: 'JetSet',
            headerTitleStyle: {
                flex: 1,
            }
        };
    };

    componentDidMount() {
        this.checkForUserData();
    }

    checkForUserData = async () => {

        const checkUser = await FileSystem.getInfoAsync(FileSystem.documentDirectory + "NewUserQuestions.json").then(
            function (result) {
                return result.exists;
            },
            function (err) {
                console.log("File fetch error: " + err);
                return false;
            }
        )

        if (checkUser == true) {
            this.props.navigation.navigate('AppStack');
        } else {
            this.props.navigation.navigate('NewUserStack');
        }
    }

    render() {
        const { navigate } = this.props.navigation;

        return (
            <View style={styles.background}>
                <View>
                    <ActivityIndicator />
                    <StatusBar barStyle="default" />
                </View>
            </View>
        );
    }
}
