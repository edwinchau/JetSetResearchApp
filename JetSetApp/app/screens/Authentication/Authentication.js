import React, { Component } from 'react';
import {
    ActivityIndicator,
    StatusBar,
    Text,
    StyleSheet,
    View,
    Button
} from 'react-native';

import styles from './Styles'
import SaveData from '../../resources/SaveData';
import * as FileSystem from "expo-file-system";

export default class Authentication extends Component {
    static navigationOptions = () => {
        return {
            headerStyle: styles.headerStyle,
            headerTintColor: '#fff',
            headerTitle: 'JetSet',
            headerTitleStyle: {
                flex: 1,
            }
        };
    };

    componentDidMount() {
        this.checkForUserData();
    }

    checkForUserData = async () => {

        const checkUser = await FileSystem.getInfoAsync(FileSystem.documentDirectory + "NewUserQuestions.json").then(
            function (result) {
                return result.exists;
            },
            function (err) {
                console.log("File fetch error: " + err);
                return false;
            }
        )

        if (checkUser == true) {
            this.props.navigation.navigate('AppStack');
        } else {
            this.props.navigation.navigate('NewUserStack');
        }
    }

    render() {
        const { navigate } = this.props.navigation;

        return (
            <View style={styles.background}>
                <View>
                    <ActivityIndicator />
                    <StatusBar barStyle="default" />
                </View>
            </View>
        );
    }
}
