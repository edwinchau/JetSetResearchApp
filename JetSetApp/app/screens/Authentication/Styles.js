import { StyleSheet } from 'react-native';

let styles = StyleSheet.create({
    background: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(108,48,237,1)',
    },
    container: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'white',
        elevation: 20,
        borderRadius: 10,
    },
    questionText: {
        marginBottom: 20,
        fontSize: 20
    },
    headerStyle: {
        backgroundColor: 'rgba(141,196,63,1)',
        height: 40,
        elevation: 5,
    },
});

module.exports = styles;
