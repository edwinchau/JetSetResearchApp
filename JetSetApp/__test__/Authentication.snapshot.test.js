import React from "react";
import Authentication from "../app/screens/Authentication"
import renderer from "react-test-renderer"

describe("<Authentication/>", ()=>{
  it("snapshot", async ()=>{
    expect(renderer.create(<Authentication/>).toJSON()).toMatchSnapshot();
  });
});
