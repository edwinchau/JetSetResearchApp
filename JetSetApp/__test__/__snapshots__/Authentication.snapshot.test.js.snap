// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authentication/> snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
